### PR TITLE
Fix Form Submit for Questions and Posts

### PR DESCRIPTION
**Pull request by Discordius**

### CODE REVIEW

**Comment by jimrandomh** [comment]
Fixed #1629 (maybe).

File: packages/lesswrong/components/posts/PostSubmit.jsx [hunk 1]
```diff
@@ -92,40 +51,15 @@ class PostSubmit extends PureComponent {
 
   render() {
     const { submitLabel = "Submit", cancelLabel = "Cancel", cancelCallback, document, collectionName, classes, router } = this.props
-
+    const { submitToFrontpage } = this.state
```
Comment (jimrandomh) [inline packages/lesswrong/components/posts/PostSubmit.jsx]: This piece of state is initialized and used, but never modified.

**Comment by Raemon** [comment]
Test cases I've thought of:

– Writing a post and posting it (set submitToFrontpage: true)
– write a post, post it (set: submitToFrontpage: false)
– write a post, drafting it, then posting it (setting submitToFrontpage: true)
– write a post, draft it, post it (submitToFrontpage: false)

– write a question, post it, (submitToFrontpage: true)
– write a question, post it, (submitToFrontpage: false)
– write a question, draft, submit (submitToFrontpage: true)
– write a question, draft, submit (submitToFrontpage: false)

– write an event, post it, (submitToFrontpage: true)
– write an event, post it, (submitToFrontpage: false)
– write an event, draft, submit (submitToFrontpage: true)
– write an event, draft, submit (submitToFrontpage: false)

**Comment by Raemon** [comment]
...and... it fails the very first test. :P 

(and the first test caused me to realize that the event tests weren't oriented correctly, they should all result in "no frontpage" which wasn't what I was thinking when I wrote them)

Currently, the form *only* shows the SubmitToFrontpage component on events, which is the opposite of what you want.

I am working on a fix

**Comment by Raemon** [comment]
Also... fails the first test after I fix the first obvious issue of "the checkbox doesn't exist", which is that *unless* you actually toggle the checkbox off-and-on-again, submitToFrontpage is undefined rather than true.

The obvious solutions to this all seemed kinda sketchy so am punting this to Tomorrow for Oli to deal with.

**Pull request merged**